neo4j/_async/driver.py

```diff
@@ -416,14 +421,20 @@ async def query(self, query, parameters=None, **kwargs):
         :returns: a new :class:`neo4j.QueryResult` object
         :rtype: QueryResult
         """
-        session_kwargs = {}
-        if "database" in kwargs:
-            session_kwargs["database"] = kwargs.pop("database")
 
-        async with self.session(**session_kwargs) as session:
-            return await session.query(query, parameters, **kwargs)
+        async with self.session(database=database) as session:
+            return await session.query(
+                query,
+                parameters,
```

Comment:
```suggestion
                parameters=parameters,
```

That's generally more future-proof.